src/pages/AddPatient.jsx

```diff
@@ -5,237 +5,303 @@ import {
   CardContent,
   Modal,
   TextField,
-  Typography,
+  Typography
 } from "@mui/material";
 import Grid from "@mui/material/Grid2";
-function AddPatient({ open, onClose }) {
+import { addPatient } from "../services/patients";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useForm } from "react-hook-form";
+import { useState } from "react";
+import Toast from "../components/Toast";
+const AddPatient = ({ open, onClose }) => {
+  const { register, handleSubmit, reset } = useForm();
+  const [openToast, setOpenToast] = useState(false);
+  const [toastMessage, setToastMessage] = useState("");
+  const [toastSeverity, setToastSeverity] = useState("");
+  const queryClient = useQueryClient();
+  function handleReset() {
+    reset();
+  }
+  const { mutate, isPending } = useMutation({
+    mutationFn: addPatient,
+    onError: (error) => {
+      setToastSeverity("error");
+      setToastMessage(`${error.message}`);
+      setOpenToast(true);
+    },
+    onSuccess: (data) => {
+      setToastSeverity("success");
+      setToastMessage(data);
+      setOpenToast(true);
+      handleReset();
+      const timeout = setTimeout(() => onClose(), 3000);
```

Comment:
There should be no need to manually set a timer to close the toast message. If you're using the snack bar, there should be a configuration that allows you to determine how long it appears on the screen